pallets/market/src/tests.rs

```diff
@@ -63,3 +67,148 @@ fn calculate_region_price_works() {
 		);
 	});
 }
+
+#[test]
+fn list_region_works() {
+	new_test_ext().execute_with(|| {
+		let region_id = RegionId { begin: 0, core: 0, mask: CoreMask::complete() };
+		let seller = 2;
+		let signer = RuntimeOrigin::signed(seller);
+
+		assert!(Regions::regions(&region_id).is_none());
+		assert_ok!(Regions::mint_into(&region_id.into(), &seller));
+
+		let record: RegionRecord<u64, u64> = RegionRecord { end: 8, owner: 1, paid: None };
+		let timeslice: u64 = <Test as crate::Config>::TimeslicePeriod::get();
+		let price = 1_000_000;
+		let recipient = 1;
+
+		// Failure: Unknown region
+
+		assert_err!(
```

Comment:
We should use `assert_noop!` intead of `assert_err!`. The difference is that `assert_noop` ensures that the state remains the same in case of an error.